rplugin/python3/deoplete/sources/deoplete_jedi/cache.py

```diff
@@ -145,39 +145,33 @@ def reap_cache(max_age=300):
     Module level completions are exempt from reaping.  It is assumed that
     module level completions will have a key length of 1.
     """
-    with _cache_lock:
-        now = time.time()
-        cur_len = len(_cache)
-        for cached in list(_cache.values()):
-            if cached.key[-1] not in ('package', 'local', 'boilerplate~',
-                                      'import~') \
-                    and now - cached._touched > max_age:
-                _cache.pop(cached.key)
-        return len(_cache), cur_len
+    while True:
+        time.sleep(300)
```

Comment:
Probably fine like it is: you might want to have `max_age=30`, but it should still be cleared every 5 minutes only.